slamd/discovery/processing/discovery_service.py

```diff
@@ -123,3 +127,23 @@ def _initialize_experiment(cls, dataframe, request_body):
             apriori_thresholds=apriori_thresholds,
             apriori_max_or_min=apriori_max_or_min,
         )
+
+    @classmethod
+    def create_tsne_plot(cls):
+        tsne_plot_data = DiscoveryPersistence.get_session_tsne_plot_data()
+        if not tsne_plot_data:
+            raise PlotDataNotFoundException("Cannot find data to create plot!")
```

Comment:
Error message should be more specific: _TSNE_ plot

Reply:
Makes sense. I will update the message.